src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/SampleService.java

```diff
@@ -21,50 +100,167 @@
    * @return the total number of sample units in the collection exercise.
    * @throws CTPException when collection exercise state transition error
    */
-  SampleUnitsRequestDTO requestSampleUnits(UUID id) throws CTPException;
+  public SampleUnitsRequestDTO requestSampleUnits(final UUID id) throws CTPException {
+    CollectionExercise collectionExercise = collectRepo.findOneById(id);
+
+    if (collectionExercise == null) {
+      throw new IllegalArgumentException(String.format("Collection exercise %s not found", id));
+    }
+
+    List<SampleLink> sampleLinks = sampleLinkRepository.findByCollectionExerciseId(id);
+    List<UUID> sampleSummaryIdList =
+        sampleLinks.stream().map(SampleLink::getSampleSummaryId).collect(Collectors.toList());
+
+    // Pre-grab and save the total number of sample units we expect to receive from the sample
+    // service BEFORE it starts to send them, to ensure no race condition
+    SampleUnitsRequestDTO responseDTO = sampleSvcClient.getSampleUnitCount(sampleSummaryIdList);
+    collexSampleCountUpdater.updateSampleSize(id, responseDTO.getSampleUnitsTotal());
+
+    // Request the sample units. They'll start arriving as soon as this line executes. Be ready!
+    SampleUnitsRequestDTO replyDTO = sampleSvcClient.requestSampleUnits(collectionExercise);
+
+    if (replyDTO != null && replyDTO.getSampleUnitsTotal() > 0) {
+      for (SampleLink samplelink : sampleLinks) {
+        partySvcClient.linkSampleSummaryId(
+            samplelink.getSampleSummaryId().toString(), collectionExercise.getId().toString());
+      }
+
+      collectionExercise.setSampleSize(replyDTO.getSampleUnitsTotal());
+      collectionExercise.setState(
+          collectionExerciseTransitionState.transition(
+              collectionExercise.getState(), CollectionExerciseEvent.EXECUTE));
+      collectRepo.saveAndFlush(collectionExercise);
+    }
+
+    return replyDTO;
+  }
 
   /**
-   * Save a SampleUnit
+   * Accepts the sample unit from the sample service. This checks that this is dealing with the
+   * initial creation of the sample, no additions of sample units to a sample unit group, no updates
+   * to a sample unit.
    *
-   * @param sampleUnit the SampleUnit to save.
-   * @return the SampleUnit saved.
-   * @throws CTPException when collection exercise state transition error
+   * @param sampleUnit the sample unit from the message.
+   * @return the saved sample unit.
    */
-  ExerciseSampleUnit acceptSampleUnit(SampleUnit sampleUnit) throws CTPException;
+  @Transactional(
+      propagation = Propagation.REQUIRED,
+      readOnly = false,
+      timeout = TRANSACTION_TIMEOUT)
```

Comment:
seems a bit excessive specifiy not readonly and a txn timeout, when the defaults would have been fine. In fact ALL of the things this specifies seem pointless

Reply:
Yeah particularly when `readOnly = false` is the default anyway.
I'm not going to change this in this PR as it's pretty prevalent across the repo/other repos. We can have another task to revisit how we're using transactions generally